File: app/src/main/java/bb/incognito/view/MainActivity.java
package bb.incognito.view;

import android.arch.lifecycle.Observer;
import android.arch.lifecycle.ViewModelProviders;
import android.content.Intent;
import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.design.widget.TabLayout;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.app.AppCompatDelegate;
import android.support.v7.widget.LinearLayoutManager;
import android.widget.SearchView;

import java.util.List;

import bb.incognito.R;
import bb.incognito.databinding.ActivityMainBinding;
import bb.incognito.model.GuestWithCocktails;
import bb.incognito.view.adapter.GuestAdapter;
import bb.incognito.viewModel.MainActivityVM;

public class MainActivity extends AppCompatActivity implements SearchView.OnQueryTextListener{
    private ActivityMainBinding mainActivityBinding;
    private MainActivityVM mainActivityViewModel;
    private GuestAdapter guestAdapter;
    private SearchView sv;

    static {
        AppCompatDelegate.setCompatVectorFromResourcesEnabled(true);
    }


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mainActivityViewModel = ViewModelProviders.of(this).get(MainActivityVM.class);
        mainActivityViewModel.setFragmentManager(this.getFragmentManager());
        setMainActivityBinding();

        sv.setOnQueryTextListener(this);
        mainActivityViewModel.getAllGuests().observe(this,
                guestWithCocktails -> guestAdapter.setGuests(guestWithCocktails));
    }

    void setMainActivityBinding()
    {
        mainActivityBinding = DataBindingUtil.setContentView(this, R.layout.activity_main);
        mainActivityBinding.setViewModel(mainActivityViewModel);
        mainActivityBinding.setLifecycleOwner(this);

        guestAdapter= new GuestAdapter();
        guestAdapter.itemTouchHelper.attachToRecyclerView(mainActivityBinding.list);
        mainActivityBinding.list.setAdapter(guestAdapter);
        mainActivityBinding.list.setLayoutManager(new LinearLayoutManager(this));
        sv = mainActivityBinding.search;

        mainActivityBinding.tabLayout.addOnTabSelectedListener(t);
    }

    TabLayout.OnTabSelectedListener t = new TabLayout.OnTabSelectedListener() {
        @Override
        public void onTabSelected(TabLayout.Tab tab) {
            switch (tab.getPosition())
            {
                case 0: return;
                case 1:  getBaseContext().startActivity(new Intent(getBaseContext(), CocktailListActivity.class));
            }
        }
        @Override
        public void onTabUnselected(TabLayout.Tab tab) {}
        @Override
        public void onTabReselected(TabLayout.Tab tab) {}
    };

    @Override
    public boolean onQueryTextSubmit(String s) {
        guestAdapter.getFilter().filter(s);
        return false;
    }

    @Override
    public boolean onQueryTextChange(String s) {
        guestAdapter.getFilter().filter(s);
        return false;
    }
}

File: app/src/main/java/bb/incognito/view/MainActivity.kt
package bb.incognito.view

import android.os.Bundle
import android.support.design.widget.TabLayout
import android.support.v7.app.AppCompatActivity
import android.support.v7.app.AppCompatDelegate
import kotlinx.android.synthetic.main.activity_main.*

import bb.incognito.R

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        supportFragmentManager.beginTransaction().add(R.id.container, CocktailsFragment()).commit()
        tabLayout.addOnTabSelectedListener(t)
    }

    internal var t: TabLayout.OnTabSelectedListener = object : TabLayout.OnTabSelectedListener {
        override fun onTabSelected(tab: TabLayout.Tab) {
            var transaction = supportFragmentManager.beginTransaction()
            when (tab.position) {
                0 -> transaction.replace(R.id.container, CocktailsFragment()).commit()
                1 -> transaction.replace(R.id.container, GuestsFragment()).commit()
                2 ->
                {
                    var x = CocktailsFragment()
                    x.menu = true
                    transaction.replace(R.id.container, x).commit()
                }

            }
        }
        override fun onTabUnselected(tab: TabLayout.Tab) {}
        override fun onTabReselected(tab: TabLayout.Tab) {}
    }
    companion object {
        init {
            AppCompatDelegate.setCompatVectorFromResourcesEnabled(true) // ??
        }
    }
}
